shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/postgresql/executor/SelectDatabaseExecutor.java

```diff
@@ -41,24 +41,43 @@
     
     private static final String DATABASE_NAME = "databasename";
     
+    private static final String DATNAME = "datname";
+    
+    private static final String NAME = "name";
+    
     private final Set<String> columnNames = new LinkedHashSet<>();
     
     private final SelectStatement sqlStatement;
     
     private String databaseNameAlias = DATABASE_NAME;
     
+    private boolean isQueryDatabase;
+    
     public SelectDatabaseExecutor(final SelectStatement sqlStatement, final String sql) {
         super(sql);
         this.sqlStatement = sqlStatement;
     }
     
     @Override
     protected void createPreProcessing() {
+        removeDuplicatedRow();
+        addDefaultRow();
+    }
+    
+    private void addDefaultRow() {
         LinkedList<String> schemaWithoutDataSource = ProxyContext.getInstance().getAllSchemaNames().stream()
                 .filter(each -> !hasDatasource(each)).collect(Collectors.toCollection(LinkedList::new));
         schemaWithoutDataSource.forEach(each -> getRows().addLast(getDefaultRowData(each)));
     }
     
+    private void removeDuplicatedRow() {
+        if (isQueryDatabase) {
+            List<Map<String, Object>> beRemovedRow = getRows().stream().collect(Collectors.groupingBy(each -> each.get(databaseNameAlias), Collectors.toCollection(LinkedList::new)))
```

Comment:
`toBeRemovedRow` may better than `beRemovedRow`.

Reply:
copy that 🥷